src/StructArrays.jl

```diff
@@ -25,6 +25,9 @@ function __init__()
             convert(WeakRefStrings.StringArray{S}, a)
         end
     end
+    Requires.@require CUDAnative="be33ccc6-a3ff-5ff2-a52e-74243cff1e17" begin
```

Comment:
This should probably use Adapt.jl instead of overloading `cudaconvert`

Reply:
I don't understand how I would use Adapt.jl.  Do you have an example somewhere you could point me too?  Or expand on it a little more?

Reply:
@vchuravy I just pushed a version that uses Adapt.  It seems to run, did I use it correctly?

Reply:
Adapt is tiny: better to load directly then in a Requires block (I really want to get rid of Requires eventually).

Reply:
You don't need CUDAnative directly as an example see httpss://github.com/JuliaArrays/OffsetArrays.jl/pull/57/files

Reply:
@piever I agree!  Thanks @vchuravy for the example on how to correctly use Adapt.  I fixed the commit so that CUDAnative is not needed.